flaskr/api/views.py

```diff
@@ -0,0 +1,39 @@
+from flask import abort, jsonify, current_app
+import json
+from . import app
+
+
+"""
+Return json for all courses under <spec> specialisation
+in <sem> semester (X1|S1|S2)
+"""
+@app.route('/specialisation/<spec>/<sem>')
+def specialisation(spec, sem):
+    sem = sem.upper()  # accept lower case request
+
+    db_file = app.root_path + '/specialisation/db.json'
+    # Try to read the json from the db file, fail gracefully and log an error
+    try:
+        db = json.load(open(db_file, 'r'))
+    except (json.JSONDecodeError, OSError) as err:
+        current_app.logger.error(
+            'API endpoint [/api/specialisation/%s/%s] failed when loading json from:\n%s\n%s',
+            spec, sem, db_file, err
+        )
+        abort(500)
```

Comment:
Instead of logging and throwing aborting with a 500, you can just throw the exception - it'll be logged with a traceback and the server will respond with a 500 🤔